rules/private/copy_file_private.bzl

```diff
@@ -78,9 +85,10 @@ def _ximpl(ctx):
     return _common_impl(ctx, True)
 
 _ATTRS = {
-    "src": attr.label(mandatory = True, allow_single_file = True),
-    "out": attr.output(mandatory = True),
+    "is_symlink": attr.bool(mandatory = True),
```

Comment:
It looks like the is_executable should be folded in here as another bool attribute.
Then you would not need _copy_file vs copy_xfile and the `if is_executable` at line 127.
Having two different techniques for doing this differentiation is confusing.

Reply:
AFAIK, we also have to set `is_executable = True` on the rule to allow it as tool for other actions.

Reply:
Yes, but it would be an attribute on copy_file, and you would just pass the value down from the macro to the rule. The virtually duplicated call at lines 131-150 would fold away.